charts/external-dns/templates/_helpers.tpl

```diff
@@ -70,3 +70,30 @@ The image to use
 {{- define "external-dns.image" -}}
 {{- printf "%s:%s" .Values.image.repository (default (printf "v%s" .Chart.AppVersion) .Values.image.tag) }}
 {{- end }}
+
+
+{{/*
+Keep backward compatibility on provider
+*/}}
+{{- define "external-dns.providername" -}}
+{{- if eq (typeOf .Values.provider) "string" }}
+{{- .Values.provider }}
+{{- else }}
+{{- .Values.provider.name }}
+{{- end }}
+{{- end }}
+
+{{/*
+The webhook providers
+*/}}
+{{- define "external-dns.webhook-providers" -}}
+{{- if eq (typeOf .Values.provider) "object" }}
+{{- $version := default "latest" .Values.provider.version }}
```

Comment:
Seems unnecessary to split the version into a separate value. Just make the image name include the tag.